magda-storage-api/src/MagdaMinioClient.ts

```diff
@@ -17,26 +25,41 @@ export default class MagdaMinioClient implements ObjectStoreClient {
             secretKey,
             region
         });
-        this.client.makeBucket(this.bucketName, region, (err: Error) => {
-            if (err) {
-                if (
-                    err.message ===
-                    "Your previous request to create the named bucket succeeded and you already own it."
-                ) {
-                    return console.log(
-                        "Bucket " + this.bucketName + " already exists 👍"
-                    );
-                } else {
-                    return console.log("😢 Error creating bucket: ", err);
+        this.region = region;
+        this.createBucket(this.bucketName);
+    }
+
+    createBucket(bucket: string): Promise<any> {
+        return new Promise((resolve, _reject) => {
+            return this.client.makeBucket(bucket, this.region, (err: Error) => {
+                if (err) {
+                    if (
+                        err.message ===
+                        "Your previous request to create the named bucket succeeded and you already own it."
+                    ) {
+                        return resolve({
+                            message: "Bucket " + bucket + " already exists 👍",
+                            success: false
+                        });
+                    } else {
+                        console.log("😢 Error creating bucket: ", err);
+                        return resolve({
```

Comment:
Why not just reject in the case of an error? It seems like the `message` being passed back here doesn't actually get used by the API anyway?

Reply:
Rejected the Promise :+1: 
The message is not being used at the moment, but it's possible that it might be useful later. At the moment, it helps make the `CreateBucketResponse` object consistent.